### PR TITLE
Add Support for Apple Silicon

### PR DESCRIPTION
**Pull request by dcvz**

### CODE REVIEW

File: core/koin-core/build.gradle [hunk 1]
```diff
@@ -17,16 +17,19 @@ kotlin {
...
         linuxX64()
         linuxArm32Hfp()
         linuxMips32()
         watchosArm32()
         watchosArm64()
         watchosX86()
         watchosX64()
+        watchosSimulatorArm64()
         tvosArm64()
         tvosX64()
```
Comment (russhwolf) [inline core/koin-core/build.gradle:34]:
```suggestion
        tvosX64()
        tvosSimulatorArm64()
```

File: core/koin-core/build.gradle [hunk 2]
```diff
@@ -100,13 +103,16 @@ kotlin {
...
             configure([
                     targets.macosX64,
+                    targets.macosArm64,
                     targets.watchosArm32,
                     targets.watchosArm64,
                     targets.watchosX86,
                     targets.watchosX64,
+                    targets.watchosSimulatorArm64,
                     targets.tvosArm64,
                     targets.tvosX64,
```
Comment (russhwolf) [inline core/koin-core/build.gradle:113]:
```suggestion
                    targets.tvosX64,
                    targets.tvosSimulatorArm64,
```

File: core/koin-test/build.gradle [hunk 3]
```diff
@@ -15,16 +15,19 @@ kotlin {
...
         linuxX64()
         linuxArm32Hfp()
         linuxMips32()
         watchosArm32()
         watchosArm64()
         watchosX86()
         watchosX64()
+        watchosSimulatorArm64()
         tvosArm64()
         tvosX64()
```
Comment (russhwolf) [inline core/koin-test/build.gradle:32]:
```suggestion
        tvosX64()
        tvosSimulatorArmX64()
```

**Review by russhwolf — COMMENTED** [review]
There's also a new tvos simulator target

**Comment by dcvz** [comment]
Pulled in the suggestions, thanks @russhwolf

File: core/koin-test/build.gradle [hunk 4]
```diff
@@ -15,18 +15,22 @@ kotlin {
...
+        tvosSimulatorArmX64()
```
Comment (moskvax) [inline core/koin-test/build.gradle:33]: I believe this is a typo; was causing issues for me building off this branch
```suggestion
        tvosSimulatorArm64()
```

**Comment by dcvz** [comment]
Thanks @moskvax !

File: core/koin-test/build.gradle (hunk 4 above)
Comment (russhwolf) [reply core/koin-test/build.gradle]: Oops, I really need to stop using GitHub as an IDE

**Review by arnaudgiuliani — APPROVED** [review]

**Comment by arnaudgiuliani** [comment]
Great. Can you fix conflicts @dcvz 🙏

**Comment by dcvz** [comment]
done @arnaudgiuliani !

**Comment by arnaudgiuliani** [comment]
great 👌

**Pull request merged**